src/app/detail/[task_id]/[task_name]/[user_type]/layout.tsx

```diff
@@ -1,15 +1,16 @@
 import { SecondaryBtn } from '@/components/buttons/SecondaryBtn'
 import { AppMargin, SizeofAppMargin } from '@/hoc/AppMargin'
 import { Stack, Typography } from '@mui/material'
-import { StyledBox, StyledKeyboardIcon, StyledTypography } from '../../ui/styledComponent'
+import { StyledBox, StyledKeyboardIcon, StyledTypography } from '@/app/detail/ui/styledComponent'
 import Link from 'next/link'
 import { ReactNode } from 'react'
+import { UserType } from '@/types/interfaces'
 
 export default function TaskDetailPageLayout({
   params,
   children,
 }: {
-  params: { task_id: string; task_name: string }
+  params: { task_id: string; task_name: string; user_type: UserType }
```

Comment:
`task_name` is used as search param so that we can directly render it in the server component without the detail API being used, which is usually considered as a good pattern while writing server rendered apps. 

`user_type` can also be passed as URL params but it doesn't make any difference passing it as param IMO. It is just a personal preference.